fuse_core/include/fuse_core/message_buffer_impl.h

```diff
@@ -137,14 +137,32 @@ void MessageBuffer<Message>::purgeHistory()
   {
     return;
   }
-  // Continue to remove the first entry from the buffer until we:
-  // (a) are left with only two entries, OR
-  // (b) the time delta between the beginning and end is within the buffer_length_
+
+  // Compute the expiration time carefully, as ROS can't handle negative times
   const ros::Time& ending_stamp = buffer_.back().first;
-  while ((buffer_.size() > 2)
-      && ((ending_stamp - buffer_.front().first) > buffer_length_))
+  ros::Time expiration_time;
```

Comment:
This is correct, but you could also do:

```suggestion
  const auto expiration_time = ending_stamp.toSec() > buffer_length_.toSec() ? ending_stamp - buffer_length_ : ros::Time(0);
```

So you can enforce correctness.